### PR TITLE
add maybe package

### PR DESCRIPTION
**Pull request by twpayne**
This is an initial proposal for a helper package [as requested](https://github.com/google/renameio/pull/20#issuecomment-689383206) so that #26 can be unblocked. The naming of the package is of course contentious, I'm happy to change it to whatever the maintainer prefers.

### CODE REVIEW

File: maybe/maybe_unix.go [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+import (
+	"io/ioutil"
+	"os"
+	"runtime"
+
+	"github.com/google/renameio"
+)
+
+// WriteFile mirrors ioutil.WriteFile. On Linux it replaces an existing file
+// with the same name atomically. On Windows the file is not written atomically
```
Comment (stapelberg) [inline maybe/maybe_unix.go:12]: How about:

On Linux it uses renameio.WriteFile to replace…

File: maybe/maybe_unix.go [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+// and most permission bits are ignored.
```
Comment (stapelberg) [inline maybe/maybe_unix.go:13]: Could you rephrase this to state “on Windows files cannot be written atomically”?

The current phrasing doesn’t convey that this is a technical limitation.

Ideally, also add a reference to the discussion where this was explained.

File: maybe/maybe_unix.go [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+func WriteFile(filename string, data []byte, perm os.FileMode) error {
```
Comment (stapelberg) [inline maybe/maybe_unix.go:14]: Could you add a paragraph stating when it is appropriate to use this function? Something like:

Prefer using renameio.WriteFile instead so that you get an error if atomic replacement is not possible on the runtime platform. maybe.WriteFile is meant as a convenience wrapper if you are okay with atomic replacement not being supported by the runtime platform.

File: maybe/maybe_unix.go [hunk 4]
```diff
@@ -0,0 +1,19 @@
+package maybe
```
Comment (stapelberg) [inline maybe/maybe_unix.go:1]: We’ll also need a package comment, but I’m assuming you would have added it after we reach agreement on the individual identifier comments…? :)

**Review by stapelberg — CHANGES_REQUESTED** [review]
Thanks for the PR, and sorry for taking so long to reply.

File: maybe/maybe_unix.go (hunk 1 above)
Comment (twpayne) [reply maybe/maybe_unix.go]: Done.

File: maybe/maybe_unix.go (hunk 3 above)
Comment (twpayne) [reply maybe/maybe_unix.go:25]: Done.

File: maybe/maybe_unix.go (hunk 4 above)
Comment (twpayne) [reply maybe/maybe_unix.go:3]: Done.

**Comment by twpayne** [comment]
Feedback addressed and commits squashed. Happy to iterate further on this if needed :)

**Review by stapelberg — APPROVED** [review]
